pyproject.toml

```diff
@@ -105,8 +101,7 @@ test = [
     "pytest-cov",
     "pytest-profiling",
     "pytest-xdist",
-    'importlib-metadata; python_version <"3.8.0"',
-    'nbmake <1.3; python_version=="3.7.*"',
+    'nbmake <1.3',
```

Comment:
You can remove the upper bound. It was only limited for Python 3.7.

Reply:
Most of the [Ruff issues](https://results.pre-commit.ci/run/github/355222824/1728569464.lR8PsRYXS9ObImXkOnA9cA) can be autofixed:

```shell
ruff check . --fix --unsafe-fixes
```